app/services/wallets/recurring_transaction_rules/validate_service.rb

```diff
@@ -40,11 +41,18 @@ def valid_threshold_trigger?
       end
 
       def valid_method?
-        (method == "target") ? valid_target_method? : true
+        (method == "target") ? valid_decimal?(params[:target_ongoing_balance]) : true
```

Comment:
In other methods you seem to return early instead, which might be clearer 🤷‍♂️ Just a suggestion

Maybe `valid_credits?` can be simplified too. `unless` combined with `||` isn't very easy, at least for me ^^

```suggestion
        return valid_decimal?(params[:target_ongoing_balance]) if method == "target"
        
        true
```